lambdas/initialise-ipv-session/src/main/java/uk/gov/di/ipv/core/initialiseipvsession/InitialiseIpvSessionHandler.java

```diff
@@ -227,6 +258,88 @@ public APIGatewayProxyResponseEvent handleRequest(
+        try {
+            // Validate
+            var inheritedIdentityJwtList =
+                    validateInheritedIdentityJwtList(inheritedIdentityJwtClaim);
+
+            var signedInheritedIdentityJWT = SignedJWT.parse(inheritedIdentityJwtList.get(0));
+            var inheritedIdentityCriConfig = configService.getCriConfig(HMRC_MIGRATION_CRI);
+
+            verifiableCredentialJwtValidator.validate(
+                    signedInheritedIdentityJWT, inheritedIdentityCriConfig, userId);
+            LogHelper.logMessage(Level.INFO, "Migration VC successfully validated");
+
+            // Store
+            try {
+                verifiableCredentialService.persistUserCredentials(
+                        signedInheritedIdentityJWT, HMRC_MIGRATION_CRI, userId);
+                ipvSessionItem.addVcReceivedThisSession(signedInheritedIdentityJWT.serialize());
+                ipvSessionService.updateIpvSession(ipvSessionItem);
+                LogHelper.logMessage(Level.INFO, "Migration VC successfully persisted");
+            } catch (VerifiableCredentialException e) {
+                throw buildRecoverableException(
+                        OAuth2Error.SERVER_ERROR
+                                .setDescription("Failed to persist inherited identity JWT - ")
+                                .appendDescription(e.getMessage()),
+                        claimsSet);
+            }
+        } catch (VerifiableCredentialException | ParseException e) {
+            throw buildRecoverableException(
+                    OAuth2Error.INVALID_REQUEST_OBJECT
+                            .setDescription("Inherited identity JWT failed to validate - ")
+                            .appendDescription(e.getMessage()),
+                    claimsSet);
+        } catch (JarValidationException e) {
+            throw buildRecoverableException(e.getErrorObject(), claimsSet);
+        }
+    }
+
+    private static List<String> validateInheritedIdentityJwtList(
+            InheritedIdentityJwtClaim inheritedIdentityJwtClaim) throws JarValidationException {
+        var inheritedIdentityJwtList =
+                Optional.of(inheritedIdentityJwtClaim.value())
+                        .orElseThrow(
+                                () ->
+                                        new JarValidationException(
+                                                OAuth2Error.INVALID_REQUEST_OBJECT.setDescription(
+                                                        "Inherited identity jwt claim received but value is null")));
+        if (inheritedIdentityJwtList.size() != 1) {
+            throw new JarValidationException(
+                    OAuth2Error.INVALID_REQUEST_OBJECT.setDescription(
+                            String.format(
+                                    "%d inherited identity jwts received - one expected",
+                                    inheritedIdentityJwtList.size())));
+        }
+        return inheritedIdentityJwtList;
+    }
```

Comment:
Given that this method validates that the list contains exactly one element, should it just return that directly?